vendor/lowrisc_opentitan/util/reggen/reg_top.sv.tpl

```diff
@@ -116,7 +116,7 @@ module ${mod_name} \
   // register signals
   logic           reg_we;
   logic           reg_re;
-  logic [AW-1:0]  reg_addr;
+  logic [BlockAw-1:0]  reg_addr;
```

Comment:
Could you elaborate on the issue for future reference sake?

Reply:
This, together with the change on line 319, ensures the correct bits are used in the selection for clarity rather than relying on the automatically inferred selection to use the lowest bits.